contracts/contract-abi-eth.json

```diff
@@ -271,7 +271,7 @@
"address"},{"name":"_owner","type":"address"},{"name":"_oracle","type":"address"},{"name":"_price","type":"uint256"},{"name":"_issuers","type":"address[]"},{"name":"_oldHavven","type":"address"}],"payable":false,"stateMutability":"nonpayable","type":"constructor"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newPrice","type":"uint256"},{"indexed":false,"name":"timestamp","type":"uint256"}],"name":"PriceUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newRatio","type":"uint256"}],"name":"IssuanceRatioUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"timestamp","type":"uint256"}],"name":"FeePeriodRollover","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"duration","type":"uint256"}],"name":"FeePeriodDurationUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"account","type":"address"},{"indexed":false,"name":"value","type":"uint256"}],"name":"FeesWithdrawn","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newOracle","type":"address"}],"name":"OracleUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newNomin","type":"address"}],"name":"NominUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newEscrow","type":"address"}],"name":"EscrowUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"account","type":"address"},{"indexed":true,"name":"value","type":"bool"}],"name":"IssuersUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"from","type":"address"},{"indexed":true,"name":"to","type":"address"},{"indexed":false,"name":"value","type":"uint256"}],"name":"Transfer","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"owner","type":"address"},{"indexed":true,"name":"spender","type":"address"},{"indexed":false,"name":"value","type":"uint256"}],"name":"Approval","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newTokenState","type":"address"}],"name":"TokenStateUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"proxyAddress","type":"address"}],"name":"ProxyUpdated","type":"event"},{"anonymous":false,"inputs":[],"name":"SelfDestructTerminated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"beneficiary","type":"address"}],"name":"SelfDestructed","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"selfDestructDelay","type":"uint256"}],"name":"SelfDestructInitiated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newBeneficiary","type":"address"}],"name":"SelfDestructBeneficiaryUpdated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"newOwner","type":"address"}],"name":"OwnerNominated","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"oldOwner","type":"address"},{"indexed":false,"name":"newOwner","type":"address"}],"name":"OwnerChanged","type":"event"}]'		
```

Comment:
Is there any documentation to match this change? None of the functions seems to be listed from etherscan.

Reply:
So the 0xC011A72400E58ecD99Ee497CF89E3775d4bd732F with the existing ABI is the Havven Proxy contract. The actualy functions for the smart contract are listed in the underlying contract listed here as target https://etherscan.io/address/0xc011a72400e58ecd99ee497cf89e3775d4bd732f#readContract

Any functions called that are not local will be passed to the target contract using assembly.

See the target contract and this is where the ABI is from 
https://etherscan.io/address/0xd989a04ad891528b571ef73dacaefeb0402a65b3

for more info see here
https://github.com/Havven/havven/blob/master/contracts/Proxy.sol

Reply:
Thanks for the clarification!